IntentsExtension/Actions/CompareLists.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+import Intents
+
+@MainActor
+final class CompareListsIntentHandler: NSObject, CompareListsIntentHandling {
+	func handle(intent: CompareListsIntent) async -> CompareListsIntentResponse {
+		let response = CompareListsIntentResponse(code: .success, userActivity: nil)
+
+		let list1 = intent.list1 ?? []
+		let list2 = intent.list2 ?? []
+		let inputFiles = list1 + list2
+
+		let set1 = Set(list1.map(\.data))
+		let set2 = Set(list2.map(\.data))
+
+		var set3: Set<Data>
+		switch intent.type {
+		case .subtracting1:
+			set3 = set2.subtracting(set1)
+		case .subtracting2:
+			set3 = set1.subtracting(set2)
+		case .intersection:
+			set3 = set1.intersection(set2)
+		case .symmetricDifference:
+			set3 = set1.symmetricDifference(set2)
+		case .unknown:
+			return .init(code: .failure, userActivity: nil)
+		}
+
+		response.result = set3.map { data in
+			inputFiles.first { $0.data == data }!
```

Comment:
This will crash if the lists are empty. It's generally better to gracefully handle such situations in intents. Just remove the `!` and use `.compactMap` instead of `.map`.